jwql/website/apps/jwql/data_containers.py

```diff
@@ -435,59 +434,43 @@ def get_available_suffixes(all_suffixes, return_untracked=True):
-    within the group to the ``n_match`` parameter.
-
-    For single files, ``n_match`` should be set to 1: any anomaly present for
-    the file is a current anomaly.  For groups of files, ``n_match`` should be
-    set to the total number of files in the group.  In this case, an
-    anomaly is returned as a current anomaly for the group only if it
-    is marked for every file in the group, i.e. there are ``n_match``
-    True values for that anomaly.
-
-    Note that the anomaly database may store multiple entries for
-    a rootfile, but only the last entry is considered current. Also
-    note that not every rootfile has an entry in the database: it only
-    appears if an anomaly has been marked for it at some point. The
-    ``n_match`` parameter is required as an argument for groups because
-    there is no way to determine from the anomaly table how many files
-    belong to a group.
+    for single rootfileinfo or sets of rootfileinfos in an exposure group. Group
+    anomalies are returned if they are true in every rootfileinfo in the set.
+    For single files,  any anomaly present for the file is a current anomaly.
+
 
     Parameters
     ----------
-    rootname : str
-        The rootname of interest.  May be a file rootname
-        (e.g. ``jw86600008001_02101_00001_guider2``) or an exposure
-        group root (e.g. ``jw86600008001_02101_00001``). The rootname
-        is matched as a starts-with filter on the anomaly database.
-    instrument : str
-        The instrument corresponding to the file.
-    n_match : int
-        The number of root files that must match in order to report
-        an anomaly as current.
+    rootfileinfo_set : RootFileInfo Queryset
+        A query set of 1 or more RootFileInfos of interest.
 
     Returns
     -------
     current_anomalies : list of str
-        A list of currently flagged anomalies for the given ``rootname``
+        A list of currently flagged anomalies for the given rootfileinfo_set
         (e.g. ``['snowball', 'crosstalk']``)
     """
-    table = getattr(di, '{}Anomaly'.format(JWST_INSTRUMENT_NAMES_MIXEDCASE[instrument.lower()]))
-    query = di.session.query(table).filter(table.rootname.startswith(rootname)).order_by(table.flag_date.desc())
-
-    all_records = query.data_frame.groupby('rootname').first()
+    all_anomalies = Anomalies.get_all_anomalies()
+    anomalies_set = []
     current_anomalies = []
-    if not all_records.empty:
-        for col, val in all_records.sum(axis=0, numeric_only=True).items():
-            # ignore rootname, flag_date, etc
-            if val == n_match and col in ANOMALIES_PER_INSTRUMENT:
-                current_anomalies.append(col)
+    empty_anomaly_found = False
+    for rootfileinfo in rootfileinfo_set:
+        try:
+            anomalies_set.append(rootfileinfo.anomalies.get_marked_anomalies())
+        except (ObjectDoesNotExist, AttributeError):
+            empty_anomaly_found = True
```

Comment:
This section is raising an error for me when trying to edit a group anomaly from an exposure page:

```
  File "data_containers.py", line 462, in get_current_flagged_anomalies
    for rootfileinfo in rootfileinfo_set:
TypeError: 'RootFileInfo' object is not iterable
```

Reply:
It looks like line 512 needs to be `rootfileinfo_set` instead of `rootfileinfo`

Reply:
weird, I tested that locally, will investigate, thanks!

Reply:
Clearly my local test was flawed, thanks for finding this!  Suggested fix works dev.

Reply:
Works for me locally too, now. Thanks!